lib/auth/scram.js

```diff
@@ -132,13 +183,21 @@ ScramSHA1.prototype.auth = function(server, connections, db, username, password,
     // Create a random nonce
     var nonce = crypto.randomBytes(24).toString('base64');
     // var nonce = 'MsQUY9iw0T9fx2MUEz6LZPwGuhVvWAhc'
-    var firstBare = f('n=%s,r=%s', username, nonce);
+
+    // NOTE: This is done b/c Javascript uses UTF-16, but the server is hashing in UTF-8.
+    // Since the username is not sasl-prep-d, we need to do this here.
+    const firstBare = Buffer.concat([
```

Comment:
do we want to further explicitly state the encoding as `utf-8` even though that's the default?

Reply:
Yeah, I can add that in.